packages/fullstack/src/render.rs

```diff
@@ -1,18 +1,38 @@
 //! A shared pool of renderers for efficient server side rendering.
-use crate::render::dioxus_core::NoOpMutations;
-use crate::server_context::SERVER_CONTEXT;
+use std::sync::Arc;
+use std::sync::RwLock;
+
+use serde::Serialize;
+use tokio::task::{spawn_blocking, spawn_local};
+
 use dioxus_lib::prelude::VirtualDom;
+use dioxus_lib::prelude::*;
 use dioxus_ssr::{
     incremental::{IncrementalRendererConfig, RenderFreshness, WrapBody},
     Renderer,
 };
-use serde::Serialize;
-use std::sync::Arc;
-use std::sync::RwLock;
-use tokio::task::spawn_blocking;
 
 use crate::prelude::*;
-use dioxus_lib::prelude::*;
+use crate::render::dioxus_core::NoOpMutations;
+use crate::server_context::SERVER_CONTEXT;
+
+#[cfg(not(target_arch = "wasm32"))]
+macro_rules! spawn_platform {
+    ($f:expr) => {
```

Comment:
I think this could be a function as well

Reply:
This one, too, could only be solved with macros to my knowledge.

Reply:
```rust
fn spawn_platform<Fut>(f: Fut) -> JoinHandle<Fut::Output>
where
    Fut: Future + 'static,
    Fut::Output: Send + 'static,
{
    spawn_blocking(move || {
        tokio::runtime::Runtime::new()
            .expect("couldn't spawn runtime")
            .block_on(f)
    })
    // spawn_local(f)
}
```

This implementation will result in the following error.

```
error[E0277]: `Fut` cannot be sent between threads safely
   --> packages\fullstack\src\render.rs:25:20
    |
25  |       spawn_blocking(move || {
    |       -------------- ^------
    |       |              |
    |  _____|______________within this `{closure@packages\fullstack\src\render.rs:25:20: 25:27}`
    | |     |
    | |     required by a bound introduced by this call
26  | |         tokio::runtime::Runtime::new()
27  | |             .expect("couldn't spawn runtime")
28  | |             .block_on(f)
29  | |     })
    | |_____^ `Fut` cannot be sent between threads safely
    |
note: required because it's used within this closure
   --> packages\fullstack\src\render.rs:25:20
    |
25  |     spawn_blocking(move || {
    |                    ^^^^^^^
note: required by a bound in `spawn_blocking`
   --> C:\Users\atty\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tokio-1.36.0\src\task\blocking.rs:206:28
    |
204 |     pub fn spawn_blocking<F, R>(f: F) -> JoinHandle<R>
    |            -------------- required by a bound in this function
205 |     where
206 |         F: FnOnce() -> R + Send + 'static,
    |                            ^^^^ required by this bound in `spawn_blocking`
help: consider further restricting this bound
    |
22  |     Fut: Future + 'static + std::marker::Send,
    |                           +++++++++++++++++++
```

It seems to me that `f` must be written inline with the `block_on` argument since it is a `!Send`. I think a macro would be necessary.

Reply:
@ealmloff I understand that you were able to solve the problem with a closure. Thank you very much for the learning opportunity!